src/SerializableSimpleXmlElement.php

```diff
@@ -22,25 +22,27 @@ public function serialize(): array
              * @var SerializableSimpleXmlElement $child
              */
             foreach ($this as $tag => $child) {
-                $temp = $child->serialize();
+                $serializedChild = $child->serialize();
                 $attributes = [];
 
+                // Check if node has attributes and if so add them.
                 foreach ($child->attributes() as $name => $value) {
                     $attributes["$name"] = (string) $value;
                 }
                 if (count($attributes) > 0) {
-                    $temp = array_merge($temp, [self::ATTRIBUTE_INDEX => $attributes]);
+                    $serializedChild = array_merge($serializedChild, [self::ATTRIBUTE_INDEX => $attributes]);
                 }
 
-                $array[$tag][] = $temp;
+                // add the node to the array with the tagname as key
+                $array[$tag][] = $serializedChild;
             }
         } else {
             // serialize attributes and text for a leaf-elements
-            $temp = trim((string) $this);
+            $serialize = trim((string) $this);
```

Comment:
```suggestion
            $serialized = trim((string) $this);
```

The naming `serialize` suggests that it is a function